core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -220,7 +220,8 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
         redeemScriptOpt.isDefined && redeemScriptOpt.get.redeemScript
           .isInstanceOf[WitnessScriptPubKeyV0]
 
-      isNativeV0 || isP2SHV0
+      // Only unsafe if it doesn't already have signatures
+      (isNativeV0 || isP2SHV0) && partialSignatures.isEmpty
```

Comment:
So this means if somebody else provided a signature we assume that it is safe? I.e. they are responsible for checking other outpoints on the tx?

Reply:
This is per-input, so if the input is signed, it won't need to be checked, but if you go to sign another it will still need to be checked

Reply:
Yup, when signing one must always do the extra verification, but if you wanted to verify signatures using a PSBT, you shouldn't be forced to do such verification

Reply:
so for example you could do
```
psbt.fromTransaction(utx).addSignatures(sigs).addWitnessUTXO(wutxo)
```
and then validate the sigs there

Reply:
meanwhile
```
psbt.fromTransaction(utx).addWitnessUTXO(wutxo).sign(0, key)
```
will still fail